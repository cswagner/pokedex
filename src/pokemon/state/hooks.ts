import { useCallback } from 'react'
import { PokemonState } from '.'
import { useAppDispatch, useAppSelector } from '../../state/hooks'
import { pokemonActions } from '.'

export const usePokemon: () => {
  state: PokemonState
  fetchByName: (name: string) => void
} = () => {
  const dispatch = useAppDispatch()
  const state = useAppSelector(({ pokemon }) => pokemon)

  const fetchByName = useCallback(
    (name: string) => {
      dispatch(pokemonActions.fetchByName(name))
    },
    [dispatch],
  )

  return { state, fetchByName }
}
