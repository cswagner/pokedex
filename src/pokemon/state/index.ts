import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
import { Pokemon } from '../Pokemon'
import { PokemonGateway } from '../gateways/PokemonGateway'
import { NetworkPokemonGatewayImpl } from '../gateways/NetworkPokemonGatewayImpl'

export type PokemonState =
  | { type: 'default' }
  | { type: 'loading' }
  | { type: 'error' }
  | { type: 'success'; pokemon: Pokemon }

const pokemonGateway: PokemonGateway = new NetworkPokemonGatewayImpl()

const fetchByName = createAsyncThunk(
  'pokemon/fetchByName',
  pokemonGateway.byName,
)

const slice = createSlice({
  name: 'pokemon',
  initialState: { type: 'default' } as PokemonState,
  reducers: {},
  extraReducers: builder => {
    builder
      .addCase(fetchByName.pending, () => ({ type: 'loading' }))
      .addCase(fetchByName.rejected, () => ({ type: 'error' }))
      .addCase(fetchByName.fulfilled, (_, action) => ({
        type: 'success',
        pokemon: action.payload,
      }))
  },
})

export const { reducer: pokemonReducer } = slice
export const pokemonActions = { fetchByName }
