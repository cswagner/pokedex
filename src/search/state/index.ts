import {
  createAsyncThunk,
  createListenerMiddleware,
  createSlice,
  PayloadAction,
} from '@reduxjs/toolkit'
import { SearchHistoryGateway } from '../gateways/SearchHistoryGateway'
import { DiskSearchHistoryGatewayImpl } from '../gateways/DiskSearchHistoryGatewayImpl'

interface SearchState {
  query: string
  history: string[]
  persistedHistory?: string[]
  isHistoryEnabled: boolean
}

const historyGateway: SearchHistoryGateway = new DiskSearchHistoryGatewayImpl()
const fetchHistory = createAsyncThunk(
  'search/fetchHistory',
  historyGateway.history,
)

const slice = createSlice({
  name: 'search',
  initialState: {
    query: '',
    history: [],
    isHistoryEnabled: false,
  } as SearchState,
  reducers: {
    updateQuery: (state, action: PayloadAction<string>) => {
      state.query = action.payload
    },
    addQueryToHistory: (state, action: PayloadAction<string>) => {
      state.history.push(action.payload)
    },
    toggleHistory: state => {
      state.isHistoryEnabled = !state.isHistoryEnabled
    },
  },
  extraReducers: builder => {
    builder
      .addCase(fetchHistory.fulfilled, (state, action) => {
        state.persistedHistory = action.payload
      })
      .addCase(fetchHistory.rejected, state => {
        state.persistedHistory = []
      })
  },
})

const listenerMiddleware = createListenerMiddleware()
listenerMiddleware.startListening({
  actionCreator: slice.actions.addQueryToHistory,
  effect: action => {
    historyGateway.add(action.payload)
  },
})

export const { reducer: searchReducer } = slice
export const { middleware: searchMiddleware } = listenerMiddleware
export const searchActions = { ...slice.actions, fetchHistory }
