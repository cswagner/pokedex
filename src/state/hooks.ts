import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { AppState, AppDispatch } from '.'

export const useAppDispatch = () => useDispatch<AppDispatch>()
export const useAppSelector: TypedUseSelectorHook<AppState> = useSelector
