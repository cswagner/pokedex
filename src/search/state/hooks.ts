import { useCallback } from 'react'
import { useAppDispatch, useAppSelector } from '../../state/hooks'
import { searchActions } from '.'

export const useSearch: () => {
  query: string
  updateQuery: (text: string) => void
  history: string[]
  isHistoryEnabled: boolean
  addQueryToHistory: (text: string) => void
  toggleHistory: () => void
} = () => {
  const dispatch = useAppDispatch()
  const query = useAppSelector(state => state.search.query)
  const history = useAppSelector(state => state.search.history)
  const isHistoryEnabled = useAppSelector(
    state => state.search.isHistoryEnabled,
  )

  const updateQuery = useCallback(
    (text: string) => {
      dispatch(searchActions.updateQuery(text))
    },
    [dispatch],
  )

  const addQueryToHistory = useCallback(
    (text: string) => {
      dispatch(searchActions.addQueryToHistory(text))
    },
    [dispatch],
  )

  const toggleHistory = useCallback(() => {
    dispatch(searchActions.toggleHistory())
  }, [dispatch])

  return {
    query,
    updateQuery,
    history,
    isHistoryEnabled,
    addQueryToHistory,
    toggleHistory,
  }
}
