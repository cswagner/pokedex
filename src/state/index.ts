import { combineReducers, configureStore } from '@reduxjs/toolkit'
import { searchReducer } from '../search/state'

const reducer = combineReducers({
  search: searchReducer,
})

export const setupStore = (preloadedState?: AppState) =>
  configureStore({ reducer, preloadedState })

export type AppState = ReturnType<typeof reducer>
export type AppStore = ReturnType<typeof setupStore>
export type AppDispatch = AppStore['dispatch']
