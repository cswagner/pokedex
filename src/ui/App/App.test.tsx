import React from 'react'
import 'react-native'
import { act, fireEvent, render } from '@testing-library/react-native'
import { App } from '.'
import { setupStore } from '../../state'

jest.useFakeTimers()

describe('App', () => {
  describe('search', () => {
    describe('on success', () => {
      const response = {
        id: '123',
        name: 'Some Name',
      }

      beforeEach(() => {
        fetchMock.mockResponse(() =>
          delay(300).then(() => JSON.stringify(response)),
        )
      })

      it('should load Pokemon', async () => {
        const store = setupStore({
          search: { query: '' },
          pokemon: { type: 'default' },
        })
        const rendered = render(
          <App
            store={store}
            searchQueryInputTestID="search-query-input"
            searchButtonTestID="search-button"
          />,
        )

        expect(rendered.toJSON()).toMatchSnapshot('default')

        const searchQueryInputView = await rendered.findByTestId(
          'search-query-input',
        )
        const searchButtonView = await rendered.findByTestId('search-button')

        fireEvent(searchQueryInputView, 'changeText', 'Some query')
        fireEvent(searchButtonView, 'press')

        await advanceTimeBy(200)

        expect(rendered.toJSON()).toMatchSnapshot('loading')

        await advanceTimeBy(200)

        expect(rendered.toJSON()).toMatchSnapshot('success')
      })
    })

    describe('on error', () => {
      beforeEach(() => {
        fetchMock.mockReject(() =>
          delay(300).then(() => {
            throw new Error('Some error')
          }),
        )
      })

      it('should show an error', async () => {
        const store = setupStore({
          search: { query: '' },
          pokemon: { type: 'default' },
        })
        const rendered = render(
          <App
            store={store}
            searchQueryInputTestID="search-query-input"
            searchButtonTestID="search-button"
          />,
        )

        expect(rendered.toJSON()).toMatchSnapshot('default')

        const searchQueryInputView = await rendered.findByTestId(
          'search-query-input',
        )
        const searchButtonView = await rendered.findByTestId('search-button')

        fireEvent(searchQueryInputView, 'changeText', 'Some query')
        fireEvent(searchButtonView, 'press')

        await advanceTimeBy(200)

        expect(rendered.toJSON()).toMatchSnapshot('loading')

        await advanceTimeBy(200)

        expect(rendered.toJSON()).toMatchSnapshot('error')
      })
    })
  })
})

const delay: (duration: number) => Promise<void> = duration =>
  new Promise(resolve => setTimeout(resolve, duration))

const advanceTimeBy: (duration: number) => Promise<void> = duration =>
  act(async () => {
    jest.advanceTimersByTime(duration)
  })
