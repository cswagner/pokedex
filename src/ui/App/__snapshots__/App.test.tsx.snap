// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App search on error should show an error: default 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "red",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "blue",
            "borderColor": "lightgray",
            "borderRadius": 36,
            "borderWidth": 2,
            "height": 72,
            "width": 72,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "red",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "yellow",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "green",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "alignItems": "center",
          "padding": 16,
          "rowGap": 16,
        }
      }
      keyboardDismissMode="on-drag"
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.25)",
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#333333",
              "borderColor": "lightgray",
              "borderWidth": 16,
              "justifyContent": "center",
              "width": "75%",
            }
          }
        />
        <Text
          style={
            {
              "backgroundColor": "#333333",
              "color": "white",
              "padding": 16,
              "textAlign": "center",
              "width": "75%",
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <TextInput
        autoCorrect={false}
        defaultValue=""
        inputMode="search"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        returnKeyType="search"
        selectionColor="black"
        style={
          [
            {
              "flex": 1,
            },
            {
              "backgroundColor": "#52ae5f",
              "color": "black",
              "padding": 16,
            },
          ]
        }
        testID="search-query-input"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="search-button"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#333333",
              "borderRadius": 24,
              "height": 48,
              "justifyContent": "center",
              "width": 48,
            }
          }
        >
          <SvgMock
            height={24}
            width={24}
          />
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`App search on error should show an error: error 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "red",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "blue",
            "borderColor": "lightgray",
            "borderRadius": 36,
            "borderWidth": 2,
            "height": 72,
            "width": 72,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "red",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "yellow",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "green",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "alignItems": "center",
          "padding": 16,
          "rowGap": 16,
        }
      }
      keyboardDismissMode="on-drag"
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.25)",
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#333333",
              "borderColor": "lightgray",
              "borderWidth": 16,
              "justifyContent": "center",
              "width": "75%",
            }
          }
        >
          <SvgMock
            height={48}
            width={48}
          />
        </View>
        <Text
          style={
            {
              "backgroundColor": "#333333",
              "color": "white",
              "padding": 16,
              "textAlign": "center",
              "width": "75%",
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <TextInput
        autoCorrect={false}
        defaultValue="Some query"
        inputMode="search"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        returnKeyType="search"
        selectionColor="black"
        style={
          [
            {
              "flex": 1,
            },
            {
              "backgroundColor": "#52ae5f",
              "color": "black",
              "padding": 16,
            },
          ]
        }
        testID="search-query-input"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="search-button"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#333333",
              "borderRadius": 24,
              "height": 48,
              "justifyContent": "center",
              "width": 48,
            }
          }
        >
          <SvgMock
            height={24}
            width={24}
          />
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`App search on error should show an error: loading 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "red",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "blue",
            "borderColor": "lightgray",
            "borderRadius": 36,
            "borderWidth": 2,
            "height": 72,
            "width": 72,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "red",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "yellow",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "green",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "alignItems": "center",
          "padding": 16,
          "rowGap": 16,
        }
      }
      keyboardDismissMode="on-drag"
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.25)",
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#333333",
              "borderColor": "lightgray",
              "borderWidth": 16,
              "justifyContent": "center",
              "width": "75%",
            }
          }
        >
          <ActivityIndicator
            color="white"
            size="large"
          />
        </View>
        <Text
          style={
            {
              "backgroundColor": "#333333",
              "color": "white",
              "padding": 16,
              "textAlign": "center",
              "width": "75%",
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <TextInput
        autoCorrect={false}
        defaultValue="Some query"
        inputMode="search"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        returnKeyType="search"
        selectionColor="black"
        style={
          [
            {
              "flex": 1,
            },
            {
              "backgroundColor": "#52ae5f",
              "color": "black",
              "padding": 16,
            },
          ]
        }
        testID="search-query-input"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="search-button"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#333333",
              "borderRadius": 24,
              "height": 48,
              "justifyContent": "center",
              "width": 48,
            }
          }
        >
          <SvgMock
            height={24}
            width={24}
          />
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`App search on success should load Pokemon: default 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "red",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "blue",
            "borderColor": "lightgray",
            "borderRadius": 36,
            "borderWidth": 2,
            "height": 72,
            "width": 72,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "red",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "yellow",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "green",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "alignItems": "center",
          "padding": 16,
          "rowGap": 16,
        }
      }
      keyboardDismissMode="on-drag"
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.25)",
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#333333",
              "borderColor": "lightgray",
              "borderWidth": 16,
              "justifyContent": "center",
              "width": "75%",
            }
          }
        />
        <Text
          style={
            {
              "backgroundColor": "#333333",
              "color": "white",
              "padding": 16,
              "textAlign": "center",
              "width": "75%",
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <TextInput
        autoCorrect={false}
        defaultValue=""
        inputMode="search"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        returnKeyType="search"
        selectionColor="black"
        style={
          [
            {
              "flex": 1,
            },
            {
              "backgroundColor": "#52ae5f",
              "color": "black",
              "padding": 16,
            },
          ]
        }
        testID="search-query-input"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="search-button"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#333333",
              "borderRadius": 24,
              "height": 48,
              "justifyContent": "center",
              "width": 48,
            }
          }
        >
          <SvgMock
            height={24}
            width={24}
          />
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`App search on success should load Pokemon: loading 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "red",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "blue",
            "borderColor": "lightgray",
            "borderRadius": 36,
            "borderWidth": 2,
            "height": 72,
            "width": 72,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "red",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "yellow",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "green",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "alignItems": "center",
          "padding": 16,
          "rowGap": 16,
        }
      }
      keyboardDismissMode="on-drag"
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.25)",
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#333333",
              "borderColor": "lightgray",
              "borderWidth": 16,
              "justifyContent": "center",
              "width": "75%",
            }
          }
        >
          <ActivityIndicator
            color="white"
            size="large"
          />
        </View>
        <Text
          style={
            {
              "backgroundColor": "#333333",
              "color": "white",
              "padding": 16,
              "textAlign": "center",
              "width": "75%",
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <TextInput
        autoCorrect={false}
        defaultValue="Some query"
        inputMode="search"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        returnKeyType="search"
        selectionColor="black"
        style={
          [
            {
              "flex": 1,
            },
            {
              "backgroundColor": "#52ae5f",
              "color": "black",
              "padding": 16,
            },
          ]
        }
        testID="search-query-input"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="search-button"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#333333",
              "borderRadius": 24,
              "height": 48,
              "justifyContent": "center",
              "width": 48,
            }
          }
        >
          <SvgMock
            height={24}
            width={24}
          />
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`App search on success should load Pokemon: success 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "red",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "blue",
            "borderColor": "lightgray",
            "borderRadius": 36,
            "borderWidth": 2,
            "height": 72,
            "width": 72,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "red",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "yellow",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "green",
            "borderColor": "lightgray",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "alignItems": "center",
          "padding": 16,
          "rowGap": 16,
        }
      }
      keyboardDismissMode="on-drag"
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.25)",
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#333333",
              "borderColor": "lightgray",
              "borderWidth": 16,
              "justifyContent": "center",
              "width": "75%",
            }
          }
        />
        <Text
          style={
            {
              "backgroundColor": "#333333",
              "color": "white",
              "padding": 16,
              "textAlign": "center",
              "width": "75%",
            }
          }
        >
          #123 SOME NAME
        </Text>
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "columnGap": 16,
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <TextInput
        autoCorrect={false}
        defaultValue="Some query"
        inputMode="search"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        returnKeyType="search"
        selectionColor="black"
        style={
          [
            {
              "flex": 1,
            },
            {
              "backgroundColor": "#52ae5f",
              "color": "black",
              "padding": 16,
            },
          ]
        }
        testID="search-query-input"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="search-button"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#333333",
              "borderRadius": 24,
              "height": 48,
              "justifyContent": "center",
              "width": 48,
            }
          }
        >
          <SvgMock
            height={24}
            width={24}
          />
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
